### PR TITLE
Implemented plugins as a struct instead of cell

### PR DESCRIPTION
**Pull request by GogiPuttar**
Plugins can now be accessed using arm.plugin.impedanceController where "impedanceController" is the "name" field of the plugin from the config file

### CODE REVIEW

**Comment by GogiPuttar** [comment]
Also added a getPluginByType method on the arm just in case someone does not know / care about the plugin's name

**Review by daverolli — APPROVED** [review]
We'll try them out and figure out which one we like better

File: include/hebi/HebiArm.m [hunk 1]
```diff
@@ -231,6 +235,22 @@
             % Send state to module
             this.group.send(this.cmd, varargin{:});
         end
+
+        function plugin = getPluginByType(this, type)
```
Comment (ennerf) [inline include/hebi/HebiArm.m:239]: There are defined `plugin` types on [robot-config docs](https://github.com/HebiRobotics/robot-config?tab=readme-ov-file#plugins-optional) that match the 

e.g.
* GravityCompensationEffort
* DynamicsCompensationEffort
* etc.

It should compare against those and not against the arbitrary MATLAB type

File: include/hebi/HebiArm.m [hunk 2]
```diff
@@ -210,8 +210,12 @@
 
             % Call plugins (FK, Jacobians, End-Effector XYZ, etc.)
             this.state = newState;
-            for i=1:length(this.plugins)
-                this.plugins{i}.update(this);
+            pluginFields = fieldnames(this.plugins);
```
Comment (ennerf) [inline include/hebi/HebiArm.m:213]: How expensive is this call? We should benchmark it

**Pull request merged**